source/app/platform/win/WInMain.cpp

```diff
@@ -0,0 +1,423 @@
+/*
+* This file contains WinMain and related functions
+*/
+
+#include "StdInc.h"
+
+#include "LoadingScreen.h"
+#include "ControllerConfigManager.h"
+#include "Gamma.h"
+
+#include "VideoPlayer.h"
+#include "VideoMode.h"
+#include "Input.h"
+#include "Platform.h"
+#include "WndProc.h"
+
+constexpr auto NO_FOREGROUND_PAUSE = true;
+
+// 0x746870
+void MessageLoop() {
+    MSG msg;
+    while (PeekMessageA(&msg, nullptr, 0, 0, PM_REMOVE | PM_NOYIELD)) {
+        if (msg.message == WM_QUIT) {
+            RsGlobal.quit = true;
+        } else {
+            TranslateMessage(&msg);
+            DispatchMessageA(&msg);
+        }
+    }
+}
+
+// 0x7486A0
+bool InitApplication(HINSTANCE hInstance) {
+    WNDCLASS windowClass      = { 0 };
+    windowClass.style         = CS_BYTEALIGNWINDOW;
+    windowClass.lpfnWndProc   = NOTSA_WndProc;
+    windowClass.hInstance     = hInstance;
+    windowClass.hIcon         = LoadIcon(hInstance, MAKEINTRESOURCE(IDI_MAIN_ICON));
+    windowClass.hCursor       = LoadCursor(nullptr, IDC_ARROW);
+    windowClass.lpszClassName = APP_CLASS;
+    return RegisterClass(&windowClass);
+}
+
+// 0x745560
+HWND InitInstance(HINSTANCE hInstance) {
+    RECT rect = { 0, 0, RsGlobal.maximumWidth, RsGlobal.maximumHeight };
+    AdjustWindowRect(&rect, WS_OVERLAPPEDWINDOW, false);
+
+    return CreateWindowEx(
+        0,
+        APP_CLASS,
+        RsGlobal.appName,
+        WS_OVERLAPPEDWINDOW,
+        CW_USEDEFAULT,
+        CW_USEDEFAULT,
+        rect.right - rect.left,
+        rect.bottom - rect.top,
+        nullptr,
+        nullptr,
+        hInstance,
+        nullptr
+    );
+}
+
+// 0x7468E0
+bool IsAlreadyRunning() {
+    CreateEventA(nullptr, false, true, APP_CLASS);
+    if (GetLastError() != ERROR_ALREADY_EXISTS) {
+        return false;
+    }
+
+    HWND window = FindWindowA(APP_CLASS, RsGlobal.appName);
+    if (window)
+        SetForegroundWindow(window);
+    else
+        SetForegroundWindow(PSGLOBAL(window));
```

Comment:
```suggestion
    if (HWND window = FindWindowA(APP_CLASS, RsGlobal.appName); window)
        SetForegroundWindow(window);
    else
        SetForegroundWindow(PSGLOBAL(window));
```
`window` var may be confused with macro